src/sympc/protocol/aby3/aby3.py

```diff
@@ -0,0 +1,172 @@
+"""ABY3 Protocol.
+
+ABY3 : A Mixed Protocol Framework for Machine Learning.
+https://eprint.iacr.org/2018/403.pdf
+"""
+# stdlib
+from typing import Any
+from typing import List
+from typing import Tuple
+
+# third party
+import torch
+import torchcsprng as csprng
+
+from sympc.protocol.protocol import Protocol
+from sympc.session import Session
+from sympc.tensor import MPCTensor
+from sympc.tensor import ReplicatedSharedTensor
+from sympc.tensor.tensor import SyMPCTensor
+
+gen = csprng.create_random_device_generator()
```

Comment:
Q: From the protocol I see that there is a generator shared only between party 2 and 3

Reply:
Yeah, in the p2p model, parties local generate random values using common prrs, and reveal only to parties 2 & 3 for truncation algorithm1, when we take this model to orchestrator , I found we had several round  complexity for `trunc1`,I made it a common generator, but when we switch truncation algorithm 2  ,it would be optimized.

Reply:
Discussed in slack.